pkg/restapi/operation/didexchange/didexchange.go

```diff
@@ -350,21 +351,66 @@ func (c *Operation) startClientEventListener() error {
 		return fmt.Errorf("didexchange message event registration failed: %w", err)
 	}
 
-	// auto execute the actions
+	// event listeners
 	go func() {
-		err := service.AutoExecuteActionEvent(c.actionCh)
-		if err != nil {
-			logger.Errorf("auto action event execution failed: %s", err)
+		for {
+			select {
+			case e := <-c.actionCh:
+				c.handleActionEvents(e)
+			case e := <-c.msgCh:
+				c.handleMessageEvents(e)
+			}
 		}
 	}()
 
-	go func() {
-		for e := range c.msgCh {
-			// TODO https://github.com/hyperledger/aries-framework-go/issues/551 - Integrate Message event with Webhook
-			// for now, log the messages
-			logger.Infof("message event received : type=%s", e.Type)
+	return nil
+}
+
+func (c *Operation) handleMessageEvents(e service.StateMsg) {
+	if e.Type == service.PostState {
+		var props didexchange.Event
+		switch v := e.Properties.(type) {
+		case didexchange.Event:
+			props = v
+		case error:
+			logger.Warnf("service processing failed : %s", v)
+			return
+		default:
+			logger.Errorf("event is not of DIDExchange event type")
+			return
 		}
-	}()
+
+		err := c.sendConnectionNotification(props.ConnectionID(), e.StateID)
```

Comment:
I see a way to simplify this function. You can just move these four lines inside of the `case didexchange.event` block. Then you don't need that extra `didexchange.Event var` at the top. And then you can also get rid of the `returns` in the `error` and `default` cases.

Reply:
[Done](https://github.com/hyperledger/aries-framework-go/compare/bbb4d18c35a98651aa199db0c9a11c90d8e3e99f..b7b4d6ceea82fbb36e3bf39e019d95c5cfb30c8c#diff-5213b774cf71e1cef26c7bb7c48ee94eR374).